### DISCUSSION TITLE
Monorepo - Deploying migrations in parallel?

### DISCUSSION THREAD

**Discussion by TheRealFlyingCoder** [prisma/prisma#18109] [Q&A]
So in a similar setup as described in [this discussion](https://github.com/prisma/prisma/discussions/12350)

If you have a monorepo with multiple projects referencing the same DB, how do you handle deploying migrations?

If all 3 applications could be deployed at the same time to each environment, are they fine to have an `npx prisma migrate deploy` command?

Will 2 of the 3 fail because one is already in progress? Does it matter?

**Comment by nurul3101** [comment]
Hey @TheRealFlyingCoder 👋 

Prisma makes use of [Advisory Locking](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#advisory-locking) when invoking `npx prisma migrate deploy` command.
 
This safeguard ensures that multiple commands cannot run at the same time, so in your case the other 2 commands will fail because the first command is already in progress.

**Reply by TheRealFlyingCoder** [reply]
Interesting, then in this scenario the overall deployments will fail for those 2 apps I suppose.

Unless Its a graceful failure and let's a deployment continue

**Reply by nurul3101** [reply]
Yes, I think the same. The other 2 app deployments will timeout and fail.

**Reply by hussamkhatib** [reply]
It happens even if the builds happens one after the other. My guess is that the connection is idle and not closed. We have 3 builds and have noticed that either the  2nd or 3rd build always fail.

**Comment by hussamkhatib** [comment]
Hi @TheRealFlyingCoder, did you manage to get a workaround for this ?

**Reply by TheRealFlyingCoder** [reply]
Hey @hussamkhatib I just ended up only attaching the migrate deploy command to one of apps.

I then set up the pipeline to ensure it always deploys first before the others

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
